IntegrationTests/BarrelConfig/IRtoTB/script/Makefile

```diff
@@ -0,0 +1,511 @@
+	rm -rfv Work/
+	vivado -mode batch -source ./makeProject.tcl
+
+### Make symbolic links ###
+
+common:
+	ln -sf $(INT_TESTS)/common common
+
+MemPrints:
+	ln -sf $(EMDATA)/MemPrintsBarrel MemPrints
+
+LUTs:
+	ln -sf $(EMDATA)/LUTsBarrel LUTs
+
+### Make HLS IPs ###
+
+InputRouterTop_IR_DTC_2S_1_A: $(TOP_FUNCS)/InputRouter_parameters.h $(TOP_FUNCS)/InputRouterTop.cc $(TOP_FUNCS)/InputRouterTop.h $(TRK_ALGO)/InputRouter.h
```

Comment:
Does the repetitive code in this Makefile, specifying how to compile each specific processing module, such as InputRouterTop_IR_DTC_2S_1_B, have to be written by hand? This will be painful to maintain, and means that we'll need several copies of this Makefile for different chains. Is there no way for the Makefile to determine this info itself, or for these parts of the Makefile to be written by a python script or similar?

Reply:
I'm looking into automatically generating prerequisites using "gcc", and it is definitely possible. I just need to do further tests and decide on the best way to organize things.

Reply:
A cursory glance suggests that the RHS of the dependency is the same on every line. If that is true, you could just create one set of dependencies and use that to reduce the copypasta.

i.e.,
```
IRDEPS=  $(TOP_FUNCS)/InputRouter_parameters.h \
                 $(TOP_FUNCS)/InputRouterTop.cc \                 
                 $(TOP_FUNCS)/InputRouterTop.h \
                 $(TRK_ALGO)/InputRouter.h
IputRouterTop_IR_DTC_2S_1_A: $(IRDEPS)
InputRouterTop_IR_DTC_PS10G_1_A: $(IRDEPS)
```
etc etc.

Reply:
Also, the argument to GCC to generate the dependencies is `-MD`; this spits out a .d file which you can then include into your makefile with something like
```
#
# Include the automatically generated dependency files.
#
ifneq (${MAKECMDGOALS},clean)
-include ${wildcard ${COMPILER}/*.d} __dummy__
endif
```
here ${COMPILER} is the directory that the .d and .o files are stored.

Reply:
Yes, I have learned a lot from the gmake manual over the past few days, and I think utilizing these automatically generated dependency files is the way to go.

In fact, I'm almost at the point where there are ~2 lines that are unique to the Makefile for each subchain, with all the actual rules being common. I just need a little more tinkering to get everything working.

Reply:
Let me know if you need a beta tester.

Reply:
@tomalin @pwittich I'm pretty sure I damaged my brain in the process, but I have greatly simplified the Makefiles, with all of the dependencies determined automatically. There are now just two lines that need to be updated when copying a Makefile to a new chain: the location of the top functions and the list of modules (top functions) in the chain.

Everything seems to work beautifully in my testing, and we'll see if the CI tests remain happy, but please feel free to try to break things yourself.

Reply:
I was able to run it successfully; it might be worth in the future exploring if we can make the download script (which now does a lot) faster too (i.e., parallelize parts of it...), maybe via `xargs` or something.

Reply:
Thanks for testing. I've made a few additional tweaks so that hopefully `make` plays nicely with the CI tests now. We'll see if my understanding was correct later this evening.

And agreed about making download.sh faster. I've had really good results using `xargs` to parallelize scripts in other projects.